src/interface/obsidian/src/utils.ts

```diff
@@ -31,39 +31,54 @@ function fileExtensionToMimeType (extension: string): string {
 export async function updateContentIndex(vault: Vault, setting: KhojSetting, lastSyncedFiles: TFile[], regenerate: boolean = false): Promise<TFile[]> {
     // Get all markdown, pdf files in the vault
     console.log(`Khoj: Updating Khoj content index...`)
-    const files = vault.getFiles().filter(file => file.extension === 'md' || file.extension === 'pdf');
-    const binaryFileTypes = ['pdf', 'png', 'jpg', 'jpeg']
+    const files = vault.getFiles().filter(file => file.extension === 'md' || file.extension === 'markdown' || file.extension === 'pdf');
+    const binaryFileTypes = ['pdf']
     let countOfFilesToIndex = 0;
     let countOfFilesToDelete = 0;
 
     // Add all files to index as multipart form data
-    const formData = new FormData();
+    const fileData = [];
     for (const file of files) {
         countOfFilesToIndex++;
         const encoding = binaryFileTypes.includes(file.extension) ? "binary" : "utf8";
         const mimeType = fileExtensionToMimeType(file.extension) + (encoding === "utf8" ? "; charset=UTF-8" : "");
         const fileContent = encoding == 'binary' ? await vault.readBinary(file) : await vault.read(file);
-        formData.append('files', new Blob([fileContent], { type: mimeType }), file.path);
+        fileData.push({blob: new Blob([fileContent], { type: mimeType }), path: file.path});
```

Comment:
Why the change in the data format?

Reply:
There's no change in data format sent as request payload to API. `fileData` variable is just a temporary variable collecting file content and path as a key, value pairs object for batch processing later. Before the HTTP request is sent, the data collected in the `fileData` is used to form the payload with `'files'` as the key (like previously)